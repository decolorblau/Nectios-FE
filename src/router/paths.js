const paths = {
  home: "/",
  detail: "/products/:id",
  login: "/login",
  newProduct: "/new",
  user: "/user",
  notFound: "/:pathMatch(.*)*",
};

export default paths;
