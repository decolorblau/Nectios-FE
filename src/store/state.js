const state = {
  products: [],
  comments: [],
  user: {
    isAuthenticated: false,
    clientKey: "",
    user: {
      email: "",
      name: "",
      profilePhoto: "",
      role: "",
      surname: "",
    },
  },
  currentProduct: {
    product: {
      id: "",
      client_id: "",
      description: "",
      key: "",
      name: "",
      price: "",
      product_photo: "",
    },
    comments: [],
  },
};

export default state;
