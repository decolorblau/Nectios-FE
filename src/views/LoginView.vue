<template>
  <div class="login-container">
    <Login />
  </div>
</template>

<script>
import { defineComponent } from "vue";
import { mapActions, mapGetters, mapState } from "vuex";
import Login from "../components/Login/Login.vue";

export default defineComponent({
  name: "LoginPage",

  components: {
    Login,
  },
  computed: {
    ...mapState(["user"]),
  },
  methods: {
    ...mapActions(["checkToken"]),
    ...mapGetters(["redirectToHome"]),
  },
  mounted() {
    this.checkToken();
    if (this.user.isAuthenticated) {
      this.redirectToHome();
    }
  },
});
</script>

<style lang="scss" scoped>
.login-container {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-items: center;
  background-color: rgb(250, 250, 250);
}
.blue {
  color: #207eeb;
}
</style>
