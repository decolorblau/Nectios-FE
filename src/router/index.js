import { createRouter, createWebHistory } from "vue-router";
import Home from "../views/Home.vue";
import Login from "../views/LoginView.vue";
import NotFound from "../views/NotFoundView.vue";
import paths from "./paths";
import User from "../views/UserView.vue";
import Detail from "../views/DetailView.vue";

// eslint-disable-next-line object-curly-newline
const { home, notFound, login, user, detail } = paths;

const routes = [
  {
    path: home,
    name: "Home",
    component: Home,
  },
  {
    path: login,
    name: "Login",
    component: Login,
  },
  {
    path: user,
    name: "User",
    component: User,
  },
  {
    path: detail,
    name: "Detail",
    component: Detail,
  },
  {
    path: notFound,
    name: "NotFound",
    component: NotFound,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
