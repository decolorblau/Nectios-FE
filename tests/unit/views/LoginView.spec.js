import { mount, shallowMount } from "@vue/test-utils";
import LoginView from "@/views/LoginView";
import { createStore } from "vuex";
import router from "@/router";
import state from "../mockState";

let store;
const chekTokenMock = jest.fn();
const redirectToHomeMock = jest.fn();

beforeEach(() => {
  store = createStore({
    state() {
      return state;
    },
    actions: { checkToken: chekTokenMock },
    getters: { redirectToHome: redirectToHomeMock },
  });
});

describe("Given a LoginView component", () => {
  describe("when is mounted", () => {
    test("Then the action checkToken should be called", () => {
      shallowMount(LoginView, {
        global: {
          plugins: [router, store],
        },
      });

      expect(chekTokenMock).toHaveBeenCalled();
    });
  });
});
