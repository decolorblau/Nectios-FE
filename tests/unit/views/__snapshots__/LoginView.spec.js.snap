// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginView component when is rendered Then it matches de snapshot 1`] = `
<div class="login-container">
  <v-container align="center">
    <div class="title">
      <h1>WELCOME TO BLUE<span class="blue">WORLD</span></h1>
    </div>
    <v-spacer></v-spacer>
    <v-row justify="center" class="container">
      <v-col cols="10" sm="8" md="6" lg="6" xl="4">
        <v-card class="pa-3 mt-8 login-card">
          <v-card-text class="mt-5">
            <v-text-field type="email" modelvalue="claudia@pruebatecnica.com" rules="() => {
                          /* istanbul ignore next */
                          cov_1dlrxdt9wm.f[9]++;
                          cov_1dlrxdt9wm.s[41]++;
                          return (
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[4][0]++, !!_ctx.email) ||
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[4][1]++, 'This field is required')
                          );
                        }" label="Email" required="" placeholder="example@nectios.com"></v-text-field>
            <v-text-field modelvalue="cc794dc0-6bc9-114c-9066-0242ac493029" rules="() => {
                          /* istanbul ignore next */
                          cov_1dlrxdt9wm.f[11]++;
                          cov_1dlrxdt9wm.s[43]++;
                          return (
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[6][0]++, !!_ctx.clientKey) ||
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[6][1]++, 'This field is required')
                          );
                        }" label="Client Key" required="" placeholder="ff000aa0-1bb1-222c-8888-0000aa123456"></v-text-field>
            <v-text-field modelvalue="Password1234" rules="() => {
                          /* istanbul ignore next */
                          cov_1dlrxdt9wm.f[13]++;
                          cov_1dlrxdt9wm.s[45]++;
                          return (
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[8][0]++, !!_ctx.password) ||
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[8][1]++, 'This field is required')
                          );
                        },() => {
                          /* istanbul ignore next */
                          cov_1dlrxdt9wm.f[14]++;
                          cov_1dlrxdt9wm.s[46]++;
                          return (
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[9][0]++, !!_ctx.password) &amp;&amp;
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[9][1]++, _ctx.password.length >= 8) ||
                            /* istanbul ignore next */
                            (cov_1dlrxdt9wm.b[9][2]++, 'At least 8 characters')
                          );
                        }" label="Password" placeholder="password example" required="" counter="MÃ­n 8 characters" type="password"></v-text-field>
          </v-card-text>
          <v-card-actions>
            <v-spacer></v-spacer>
            <v-btn variant="outlined" class="mb-1 mr-2" disabled="false" color="#207eeb" text=""> LOGIN </v-btn>
          </v-card-actions>
        </v-card>
        <div class="errorMessage">
          <!--v-if-->
        </div>
      </v-col>
    </v-row>
  </v-container>
</div>
`;
